Add samples documentation

avlaev opened the pull request:
Add flow diagrams, config parameters placeholders

Note: the Xcode project now won't build until you fill in the placeholder parameters in Config.m

tatiNo5 commented on WebsiteLoginSample/README.md:
````diff
@@ -1,9 +1,480 @@
+
+parseResponse extracts the information, including the base url and access code from the QR code. The extracted information is obtained as a JSON data structure and passed to the serviceReaded method for further processing.
+
+```
````
You could remove this code snippet

tatiNo5 commented on WebsiteLoginSample/README.md:
````diff
@@ -1,9 +1,480 @@
+}
+```
+
+`[self serviceReaded:jsonData accessCode:strCode];` passes the json data and access code to the serviceReaded method, which uses setBackend to set the backend url:
````
`serviceReaded`, `setBackend`

tatiNo5 replied:
enter with dot the sentence and add the next comment as part of the same paragraph.

avlaev replied:
Enter with dot? I don't understand what you mean here.

tatiNo5 commented on WebsiteLoginSample/README.md:
````diff
@@ -1,9 +1,480 @@
+`[self serviceReaded:jsonData accessCode:strCode];` passes the json data and access code to the serviceReaded method, which uses setBackend to set the backend url:
+
+```
+- ( void ) serviceReaded: (NSData *)service accessCode:(NSString *) accessCode
````
remove this code snippet

tatiNo5 commented on WebsiteLoginSample/README.md:
````diff
@@ -1,9 +1,480 @@
+}
+```
+
+The key section above is:
````
The key section in it is:

tatiNo5 commented on WebsiteLoginSample/README.md:
````diff
@@ -1,9 +1,480 @@
+
+
+```
+MpinStatus *mpinStatus = [MPinMFA SetBackend:config[@"url"] rpsPrefix:config[@"rps_prefix"]];
````
There should be no `rps_prefix`! Please, check all code snippets for relevance after the text is finalized

tatiNo5 commented on WebsiteLoginSample/README.md:
````diff
@@ -1,9 +1,480 @@
+
+```
+
+This sets the backend using the base url retrieved from the QR code, and then calls onSetBackendCompleted.
````
onSetBackendCompleted should be emphasized

tatiNo5 commented on WebsiteLoginSample/README.md:
````diff
@@ -1,9 +1,480 @@
+
+onSetBackendCompleted then checks for any users already registered with the backend:
+
+```
````
remove this code snippet

tatiNo5 commented on WebsiteLoginSample/README.md:
```diff
@@ -1,9 +1,480 @@
+
+If no users are registered then the app navigates to the User Registration page (RegisterViewController.m). If there is a user then it checks the user's state:
+
+If the user is in the STARTED_REGISTRATION state the app navigates to the finish registration stage. This is also handled by the RegisterViewController.m page.
```
list these three lines with `*` to look better in the md

tatiNo5 commented on InAppLoginSample/README.md:
```diff
@@ -1,9 +1,112 @@
+
+The demo app is only configured to deal with one registered user, so if `arrUsers.count` returns `> 1` user, the SDK `DeleteUser` method is called to delete all users. `setupAddId` will call the `AddId` view to begin the user registration process.
+
+If the result is that there is one user found in the array (`_user = arrUsers[0];`) then the SDK getState method is called:
```
emphasize: `getState`

tatiNo5 commented on InAppLoginSample/README.md:
```diff
@@ -1,9 +1,112 @@
+
+If the result is that there is one user found in the array (`_user = arrUsers[0];`) then the SDK getState method is called:
+
+If the user state is INVALID, the `DeleteUser` method is called and `setupAddId` will call `viewAddId` to begin the user registration process.
```
Format this and the following lines as list in the md... with `*` at the beginning

tatiNo5 commented on WebsiteLoginSample/README.md:
````diff
@@ -1,9 +1,480 @@
+}
+```
+
+If no users are registered then the app navigates to the User Registration page (RegisterViewController.m). If there is a user then it checks the user's state:
````
remove `then`... When no users are registered the app....

tatiNo5 commented on WebsiteLoginSample/README.md:
```diff
@@ -1,9 +1,480 @@
+
+![register-confirm](Docs/register-confirm.png)
+
+Once the user has entered their email address, the Submit button starts the registration process. Towards the end of the file, this is controlled by onClickAddmButton, where getIUserById checks to make sure that the email is not already registered with the backend. If a valid new email address is entered then MakeNewUser beings the registration process and a confirmation email is sent to the user ...
```
emphasize: `onClickAddmButton` (Is the name proper? Addm?), `getIUserById`, `MakeNewUser`
...then MakeNewUser **begins** the registration process...
End with a dot and remove the code snippet after it

avlaev replied:
onClickAddmButton is the way it is named in the code. What is the abbreviation I can't really say.

tatiNo5 commented on WebsiteLoginSample/README.md:
````diff
@@ -1,9 +1,480 @@
+}
+```
+
+Once the email has been sent and the "Your Identity has been created!" message is displayed, and the user has clicked on the confirmation link in the email, they can click on the Confirmed button, which is controlled by onClickConfirmButton. Here the ConfirmRegistration method is used to make a status check which returns OK if the confirmation link has been followed in the user's email. A client  ...
````
A message "Your Identity has been created!" is displayed on the screen and the user needs to confirm their identity from the received mail by clicking on the confirmation link. Then, back in the iOS app screen, click the `Confirmed` button which is controlled by `onClickConfirmButton`. Here the `ConfirmRegistration` method is used to make a status check if the confirmation link has been followed in the user's email. Then a `client key` is retrieved for the user and the PinPadViewController.m view is instantiated to enable the user to create their PIN.

tatiNo5 commented on WebsiteLoginSample/README.md:
````diff
@@ -1,9 +1,480 @@
+
+Once the email has been sent and the "Your Identity has been created!" message is displayed, and the user has clicked on the confirmation link in the email, they can click on the Confirmed button, which is controlled by onClickConfirmButton. Here the ConfirmRegistration method is used to make a status check which returns OK if the confirmation link has been followed in the user's email. A client  ...
+
+```
````
remove this code snippet

tatiNo5 commented on WebsiteLoginSample/README.md:
````diff
@@ -1,9 +1,480 @@
+}
+```
+
+If there is a problem (i.e. email does not arrive) the user has the option to press the Re-send email button, which is controlled by onClickResendButton. This invalidates the registration with DeleteUser and starts it again with MakeNewUser:
````
emphasize `onClickResendButton`, `DeleteUser`, `MakeNewUser` and end with a dot

tatiNo5 commented on WebsiteLoginSample/README.md:
````diff
@@ -1,9 +1,480 @@
+
+If there is a problem (i.e. email does not arrive) the user has the option to press the Re-send email button, which is controlled by onClickResendButton. This invalidates the registration with DeleteUser and starts it again with MakeNewUser:
+
+```
````
remove it

tatiNo5 commented on WebsiteLoginSample/README.md:
````diff
@@ -1,9 +1,480 @@
+
+This starts with using listUsers to create an array and check to see if any users are already registered with the backend, and displays the PIN pad for the registered user, if found (note that the demo app only allows for one identity per backend):
+
+```
````
Remove the code snippet

tatiNo5 commented on WebsiteLoginSample/README.md:
```diff
@@ -1,9 +1,480 @@
+
+![4-enterpin](Docs/4-enterpin.jpg)
+
+This starts with using listUsers to create an array and check to see if any users are already registered with the backend, and displays the PIN pad for the registered user, if found (note that the demo app only allows for one identity per backend):
```
This controller uses `listUsers` method to create....

tatiNo5 commented on WebsiteLoginSample/README.md:
````diff
@@ -1,9 +1,480 @@
+}
+```
+
+Once a PIN is entered, the user taps the Send button. Within onClickSendButton the FinishRegistration method is called if the user state is STARTED_REGISTRATION. This finalizes the user registration process. It extracts the M-Pin Token from the Client Key for the provided pin (secret), and then stores the token in the secure storage. On successful completion, the user state will be set to REGISTE ...
````
emphasize: `onClickSendButton`, `FinishRegistration`

tatiNo5 replied:
Once a PIN is entered, the user taps the Send button. Within `onClickSendButton` the `FinishRegistration` method is called if the user state is STARTED_REGISTRATION. This finalizes the user registration process. It extracts the M-Pin Token from the `client key` for the provided PIN (secret), and then stores the token in the secure storage. On successful completion, the user state will be set to REGISTERED and the method will return status OK.

tatiNo5 commented on WebsiteLoginSample/README.md:
````diff
@@ -1,9 +1,480 @@
+Once a PIN is entered, the user taps the Send button. Within onClickSendButton the FinishRegistration method is called if the user state is STARTED_REGISTRATION. This finalizes the user registration process. It extracts the M-Pin Token from the Client Key for the provided pin (secret), and then stores the token in the secure storage. On successful completion, the user state will be set to REGISTE ...
+
+```
+- (IBAction)onClickSendButton:(id)sender {
````
remove it

tatiNo5 commented on WebsiteLoginSample/README.md:
````diff
@@ -1,9 +1,480 @@
+}
+```
+
+If the user's state is REGISTERED they are able to authenticate and login using their PIN. If the pin is correct the user is successfully authenticated and the app navigates the user to the Login Successful page:
````
Finally, if the user's state is REGISTERED, the setup PIN could be used for further successful login and authentication. The app navigates the user to the Login Successful page:

tatiNo5 commented on WebsiteLoginSample/WebsiteLogin/Config.m:
```diff
@@ -0,0 +1,18 @@
+//
+//  Created by Aleksandar Vlaev on 4/17/19.
+//  Copyright © 2019 MIRACL. All rights reserved.
+//
```
remove

tatiNo5 commented on WebsiteLoginSample/WebsiteLogin/Config.m:
```diff
@@ -0,0 +1,18 @@
+
+@implementation Config
++(NSString*) clientId {
+    return <# Replace with company id #>;
```
with your company id from the platform

avlaev commented on InAppLoginSample/README.md:
```diff
@@ -1,9 +1,112 @@
 >> pod install
 
-After these steps you have to go to the desired project dir and open the .xcworkspace file. Once the Xcode is loaded you should be able to build the project and review the flow by details.
+After these steps you have to go to the desired project dir and open the .xcworkspace file. Please continue with the instructions on how to issue the needed credentials and configure the app so it can build and run.
```
You missed some words. I changed it to "Then you can navigate to the desired project's directory and open its .xcworkspace file."

avlaev commented on InAppLoginSample/README.md:
```diff
@@ -1,9 +1,112 @@
+
+### Create a demo web app to act as a backend service
+
+Now that the project is imported, in order to enable you to test the demo iOS app, you should first of all create a demo MFA web app using one of our web SDKs, as explained in the [SDK Instructions](https://devdocs.trust.miracl.cloud/mobile-sdk-instructions/) section of this documentation. The purpose of this is to have a running service which can authenticate all users who are registering with a ...
```
It is meant as written - imported, meaning that now that the project is checked out, configured and all the setup is finished the user can follow the next steps.

tatiNo5 replied:
I've written in the next comment how I think it sound better without using this word at all :)

avlaev commented on InAppLoginSample/README.md:
```diff
@@ -1,9 +1,112 @@
+
+Note that, for the web app settings in the authentication portal you will need to make sure there is a redirect url which has the private IP of the host machine as its base:
+
+![redirect-url-private-ip](Docs/redirect-url-private-ip.png)
```
I don't understand. Do you want to have the text:
"Note that, for the web app settings in the authentication portal you will need to make sure there is a redirect url which has the private IP of the host machine as its base"
also added before the link to the image?

avlaev commented on InAppLoginSample/README.md:
````diff
@@ -1,9 +1,112 @@
+
+![configm-placeholders](Docs/configm-placeholders.png)
+
+```+(NSString*) clientId``` replace with your company id.
````
This is the way the property was named before I started changing anything. So I kept the name convention. I guess the control over the naming convention(code reviews) before that was really bad.

avlaev commented on InAppLoginSample/README.md:
````diff
@@ -1,9 +1,112 @@
+
+```+(NSString*) clientId``` replace with your company id.
+
+```+(NSString*) backendDomain ``` fill in the placeholder with the private IP/domain which you filled in during the previous step(for example `192.168.1.18`)
````
I changed this although I think the original comment is much more clear and more important straightforward.

avlaev commented on InAppLoginSample/README.md:
```diff
@@ -1,9 +1,112 @@
+![ios-local-view-started-reg](Docs/ios-local-view-started-reg.png)
+
+
+### PinPadViewController.m
```
I changed it to be a subtitle of a section but I would still prefer to have it in bold style.

tatiNo5 replied:
In the next sample readme I understand what's the point of this as all controllers was made as titles and explained what are doing, but here was only this one which is why it made me confused. It's good to follow the same structure in both sample readmes. I'm leaving to you to decide what it will be.

avlaev commented on InAppLoginSample/InAppLogin/Config.m:
```diff
@@ -0,0 +1,36 @@
+}
+
++(NSString*) authCheckUrl {
+    return @"https://demo.trust.miracl.cloud/authtoken";
```
Changed it but will require testing to verify that it will work.

tatiNo5 replied:
You may need to add this endpoint to your dotnet backend. The are implemented only in the main Authentication sample of the dotnet repo, so if you need them in some of the others, you need to implement them by yourself. Ping me if you need more info about that.

tatiNo5 commented:
You need to squash your commits

tatiNo5 commented on WebsiteLoginSample/README.md:
```diff
@@ -1,9 +1,107 @@
+
+Once a PIN is entered, the user taps the Send button. Within `onClickSendButton` the `FinishRegistration` method is called if the user state is `STARTED_REGISTRATION`. This finalizes the user registration process. It extracts the M-Pin Token from the `client key` for the provided PIN (secret), and then stores the token in the secure storage. On successful completion, the user state will be set to ...
+
+If the user's state is `REGISTERED` they are able to authenticate and login using their PIN. If the pin is correct the user is successfully authenticated and the app navigates the user to the Login Successful page:
```
............. If the PIN is correct the user.......

avlaev commented on WebsiteLoginSample/README.md:
```diff
@@ -1,9 +1,107 @@
+
+To build the project you will need to replace the placeholders with the appropriate parameters:
+
+<img src="Docs/configm-placeholder.png" width="300">
```
The autzurl change is for the InappLoginSample project, so it shouldn't be reflected in the readme of the WebsiteLoginSample.

tatiNo5 replied:
The screenshot needs to be updated

avlaev commented on InAppLoginSample/README.md:
```diff
@@ -1,9 +1,106 @@
+Then you can navigate to the desired project's directory and open its .xcworkspace file. Please continue with the instructions on how to issue the needed credentials and configure the app so it can build and run.
+
+
+## [Create a demo web app to act as a backend service](#demoappsetup)
```
I tried but I don't think it worked.

tatiNo5 replied:
It will work after we merge it

avlaev commented:
I squashed the commits and added the changes. Just keep in mind that the projects won't build for simulators(because of the hotfix I did to make it build for devices).

tatiNo5 commented on InAppLoginSample/README.md:
```diff
@@ -1,9 +1,106 @@
+
+The demo app is only configured to deal with one registered user, so if `arrUsers.count` returns `> 1` users, the SDK `DeleteUser` method is called to delete all users. `setupAddId` will call the `AddId` view to begin the user registration process.
+
+* If the result is that there is one user found in the array (`_user = arrUsers[0];`) then the SDK `getState` method is called.
```
This should not be part of the list... remove the starting *, please

tatiNo5 commented on InAppLoginSample/README.md:
```diff
@@ -1,9 +1,106 @@
+
+##### PinPadViewController.m
+
+<img src="Docs/ios-local-view-enter-pin.png" width="400">
```
Can you switch the place of this image with the next sentence, so the section starts with an intro sentence

tatiNo5 commented on WebsiteLoginSample/README.md:
```diff
@@ -1,9 +1,107 @@
+
+<img src="Docs/demo-proj-navigator.png" width="300">
+
+The main storyboard view `SampleApp/MPinApp/MPinApp/Base.lproj/Main.storyboard` shows the configuration of the different views and associated messages:
```
The path and the next screenshot are not relevant to this project

avlaev commented:
The requested changes have been added.
